### PR TITLE
Sprint 1

### PR DESCRIPTION
**Pull request by mjoksa**

### CODE REVIEW

File: src/Lykke.AlgoStore.Api/Lykke.AlgoStore.Api.csproj [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
+    </None>
+  </ItemGroup>
+
+  <ItemGroup>
+    <PackageReference Include="AutoMapper" Version="6.2.2" />
+    <PackageReference Include="Lykke.Common" Version="3.0.2" />
+    <PackageReference Include="Lykke.Common.ApiLibrary" Version="1.0.20" />
+    <PackageReference Include="Lykke.Logs" Version="3.4.1" />
+    <PackageReference Include="Lykke.SettingsReader" Version="1.0.43" />
```
Comment (aushakov) [inline src/Lykke.AlgoStore.Api/Lykke.AlgoStore.Api.csproj]: Min version required Lykke.SettingsReader 1.1.4

File: src/Lykke.AlgoStore.Api/Lykke.AlgoStore.Api.csproj [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      <CopyToPublishDirectory>PreserveNewest</CopyToPublishDirectory>
+      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
+    </None>
+  </ItemGroup>
+
+  <ItemGroup>
+    <PackageReference Include="AutoMapper" Version="6.2.2" />
+    <PackageReference Include="Lykke.Common" Version="3.0.2" />
+    <PackageReference Include="Lykke.Common.ApiLibrary" Version="1.0.20" />
+    <PackageReference Include="Lykke.Logs" Version="3.4.1" />
```
Comment (aushakov) [inline src/Lykke.AlgoStore.Api/Lykke.AlgoStore.Api.csproj]: Min version required Lykke.Logs 3.5.1

File: src/Lykke.AlgoStore.Core/Domain/Repositories/IAlgoBlobReadOnlyRepository.cs [hunk 3]
```diff
@@ -0,0 +1,11 @@
+﻿using System.Threading.Tasks;
+
+namespace Lykke.AlgoStore.Core.Domain.Repositories
+{
+    public interface IAlgoBlobReadOnlyRepository
+    {
+        Task<bool> BlobExists(string blobKey);
```
Comment (aushakov) [inline src/Lykke.AlgoStore.Core/Domain/Repositories/IAlgoBlobReadOnlyRepository.cs]: BlobExistsAsync

File: src/Lykke.AlgoStore.Core/Domain/Repositories/IAlgoMetaDataReadOnlyRepository.cs [hunk 4]
```diff
@@ -0,0 +1,12 @@
+﻿using System.Threading.Tasks;
+using Lykke.AlgoStore.Core.Domain.Entities;
+
+namespace Lykke.AlgoStore.Core.Domain.Repositories
+{
+    public interface IAlgoMetaDataReadOnlyRepository
```
Comment (aushakov) [inline src/Lykke.AlgoStore.Core/Domain/Repositories/IAlgoMetaDataReadOnlyRepository.cs]: Add Async to all methods

File: src/Lykke.AlgoStore.Core/Domain/Repositories/IAlgoMetaDataRepository.cs [hunk 5]
```diff
@@ -0,0 +1,11 @@
+﻿using System.Threading.Tasks;
+using Lykke.AlgoStore.Core.Domain.Entities;
+
+namespace Lykke.AlgoStore.Core.Domain.Repositories
+{
+    public interface IAlgoMetaDataRepository : IAlgoMetaDataReadOnlyRepository
```
Comment (aushakov) [inline src/Lykke.AlgoStore.Core/Domain/Repositories/IAlgoMetaDataRepository.cs]: Add Async to all methods

File: src/Lykke.AlgoStore.Core/Domain/Repositories/IAlgoRuntimeDataReadOnlyRepository.cs [hunk 6]
```diff
@@ -0,0 +1,10 @@
+﻿using System.Threading.Tasks;
+using Lykke.AlgoStore.Core.Domain.Entities;
+
+namespace Lykke.AlgoStore.Core.Domain.Repositories
+{
+    public interface IAlgoRuntimeDataReadOnlyRepository
+    {
+        Task<AlgoClientRuntimeData> GetAlgoRuntimeData(string clientId, string algoId);
```
Comment (aushakov) [inline src/Lykke.AlgoStore.Core/Domain/Repositories/IAlgoRuntimeDataReadOnlyRepository.cs]: Add Async

File: src/Lykke.AlgoStore.Core/Domain/Repositories/IAlgoRuntimeDataRepository.cs [hunk 7]
```diff
@@ -0,0 +1,11 @@
+﻿using System.Threading.Tasks;
+using Lykke.AlgoStore.Core.Domain.Entities;
+
+namespace Lykke.AlgoStore.Core.Domain.Repositories
+{
+    public interface IAlgoRuntimeDataRepository : IAlgoRuntimeDataReadOnlyRepository
```
Comment (aushakov) [inline src/Lykke.AlgoStore.Core/Domain/Repositories/IAlgoRuntimeDataRepository.cs]: Add Async to all methods

File: src/Lykke.AlgoStore.Core/Services/IAlgoStoreClientDataService.cs [hunk 8]
```diff
@@ -0,0 +1,14 @@
+﻿using System.Threading.Tasks;
+using Lykke.AlgoStore.Core.Domain.Entities;
+
+namespace Lykke.AlgoStore.Core.Services
+{
+    public interface IAlgoStoreClientDataService
```
Comment (aushakov) [inline src/Lykke.AlgoStore.Core/Services/IAlgoStoreClientDataService.cs]: Add Async for all methods

File: src/Lykke.AlgoStore.Core/Services/IAlgoStoreService.cs [hunk 9]
```diff
@@ -0,0 +1,15 @@
+﻿using System.Threading.Tasks;
+using Lykke.AlgoStore.Core.Domain.Entities;
+
+namespace Lykke.AlgoStore.Core.Services
+{
+    public interface IAlgoStoreService
```
Comment (aushakov) [inline src/Lykke.AlgoStore.Core/Services/IAlgoStoreService.cs]: Add Async for all methods

File: src/Lykke.AlgoStore.Core/Settings/ServiceSettings/DbSettings.cs [hunk 10]
```diff
@@ -0,0 +1,8 @@
+﻿namespace Lykke.AlgoStore.Core.Settings.ServiceSettings
+{
+    public class DbSettings
+    {
+        public string LogsConnString { get; set; }
```
Comment (aushakov) [inline src/Lykke.AlgoStore.Core/Settings/ServiceSettings/DbSettings.cs]: The LogsConnectionString seems better

File: src/Lykke.AlgoStore.Core/Settings/SlackNotifications/SlackNotificationsSettings.cs [hunk 11]
```diff
@@ -0,0 +1,7 @@
+﻿namespace Lykke.AlgoStore.Core.Settings.SlackNotifications
+{
+    public class SlackNotificationsSettings
+    {
+        public AzureQueuePublicationSettings AzureQueue { get; set; }
```
Comment (aushakov) [inline src/Lykke.AlgoStore.Core/Settings/SlackNotifications/SlackNotificationsSettings.cs]: You can use Lykke.AzureQueueIntegration.AzureQueueSettings instead

File: src/Lykke.AlgoStore.Services/Lykke.AlgoStore.Services.csproj [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+  <PropertyGroup>
+    <TargetFramework>netcoreapp2.0</TargetFramework>
+    <AssemblyName>Lykke.AlgoStore.Services</AssemblyName>
+    <RootNamespace>Lykke.AlgoStore.Services</RootNamespace>
+    <Version>1.0.0</Version>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="CsvHelper" Version="6.1.0" />
+    <PackageReference Include="Flurl.Http" Version="2.1.1" />
```
Comment (aushakov) [inline src/Lykke.AlgoStore.Services/Lykke.AlgoStore.Services.csproj]: I guess you can remove Flurl.Http and CsvHelper

File: tests/Lykke.AlgoStore.Tests/Infrastructure/SettingsMock.cs [hunk 13]
```diff
@@ -0,0 +1,18 @@
...
+namespace Lykke.AlgoStore.Tests.Infrastructure
+{
+    public static class SettingsMock
+    {
+        public static IReloadingManager<string> GetSettings()
+        {
+            var reloadingMock = new Mock<IReloadingManager<string>>();
+            reloadingMock
+                .Setup(x => x.Reload())
+                .Returns(() => Task.FromResult("DefaultEndpointsProtocol=https;AccountName=algostoredev;AccountKey=d2VaBHrf8h8t622KvLeTPGwRP4Dxz9DTPeBT9H3zmjcQprQ1e+Z6Sx9RDqJc+zKwlSO900fzYF2Dg6oUBVDe1w=="));
```
Comment (aushakov) [inline tests/Lykke.AlgoStore.Tests/Infrastructure/SettingsMock.cs]: Connection string is here. It's Okay?

File: src/Lykke.AlgoStore.Api/Controllers/IsAliveController.cs [hunk 14]
```diff
@@ -0,0 +1,60 @@
...
+                    (int)HttpStatusCode.InternalServerError,
+                    new ErrorResponse { ErrorMessage = $"Service is unhealthy: {healthViloationMessage}" });
+            }
+
+            // NOTE: Feel free to extend IsAliveResponse, to display job-specific indicators
+            return Ok(new IsAliveResponse
+            {
+                Name = Microsoft.Extensions.PlatformAbstractions.PlatformServices.Default.Application.ApplicationName,
+                Version = Microsoft.Extensions.PlatformAbstractions.PlatformServices.Default.Application.ApplicationVersion,
+                Env = Environment.GetEnvironmentVariable("ENV_INFO"),
```
Comment (aushakov) [inline src/Lykke.AlgoStore.Api/Controllers/IsAliveController.cs]: You can use Program.EnvInfo

File: tests/Lykke.AlgoStore.Tests/Infrastructure/SettingsMock.cs (hunk 13 above)
Comment (NovichikhinAlexey) [reply tests/Lykke.AlgoStore.Tests/Infrastructure/SettingsMock.cs]: No remove all connection string from core.
It's must be in settings service

File: src/Lykke.AlgoStore.Core/Settings/SlackNotifications/SlackNotificationsSettings.cs (hunk 11 above)
Comment (mjoksa) [reply src/Lykke.AlgoStore.Core/Settings/SlackNotifications/SlackNotificationsSettings.cs]: Latest code in lykke.dotnettemplates still uses AzureQueuePublicationSettings so I am not sure what to do here. Can you, please, provide me with link to some of existing repos that have this already implemented and I will take a look and update code accordingly?
Thanks

File: tests/Lykke.AlgoStore.Tests/Infrastructure/SettingsMock.cs (hunk 13 above)
Comment (mjoksa) [reply tests/Lykke.AlgoStore.Tests/Infrastructure/SettingsMock.cs]: This is inside test project. I am not sure how to get (load) settings inside test project so please provide me with a link to any of existing repos that have this already implemented. If there are none :) please give some suggestions on what to do here.
Thanks

File: src/Lykke.AlgoStore.Core/Settings/SlackNotifications/SlackNotificationsSettings.cs (hunk 11 above)
Comment (aushakov) [reply src/Lykke.AlgoStore.Core/Settings/SlackNotifications/SlackNotificationsSettings.cs]: For example https://github.com/LykkeCity/Lykke.Service.Regulation/blob/master/src/Lykke.Service.Regulation/Settings/SlackNotifications/SlackNotificationsSettings.cs
https://github.com/LykkeCity/Lykke.Service.Regulation/blob/master/src/Lykke.Service.Regulation/Startup.cs#L191
It's just an advice, not critical issue.

File: src/Lykke.AlgoStore.Api/Infrastructure/Extensions/HttpContextExtensions.cs [hunk 15]
```diff
@@ -0,0 +1,50 @@
...
+            if (httpContext?.Request?.Headers?.TryGetValue(headerName, out values) ?? false)
+            {
+                string rawValues = values.ToString();   // writes out as Csv when there are multiple.
+
+                if (!string.IsNullOrEmpty(rawValues))
+                    return (T)Convert.ChangeType(values.ToString(), typeof(T));
+            }
+            return default(T);
+        }
+        public static string GetIp(this HttpContext context)
```
Comment (aushakov) [inline src/Lykke.AlgoStore.Api/Infrastructure/Extensions/HttpContextExtensions.cs]: This extension method already exist in Lykke.Common, Version=4.1.0.0 Lykke.Common.Extensions.HttpContextExtensions

File: src/Lykke.AlgoStore.Api/Infrastructure/Extensions/HttpRequestExtensions.cs [hunk 16]
```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using Microsoft.AspNetCore.Http;
+
+namespace Lykke.AlgoStore.Api.Infrastructure.Extensions
+{
+    public static class HttpRequestExtensions
+    {
+        public static string GetUserAgent(this HttpRequest request)
```
Comment (aushakov) [inline src/Lykke.AlgoStore.Api/Infrastructure/Extensions/HttpRequestExtensions.cs]: Lykke.Common, Version=4.1.0.0 Lykke.Common.Extensions.HttpContextExtensions.GetUserAgent

**Pull request merged**